docs/sphinx-docs/doc-regen/makedocumentation.py

```diff
@@ -0,0 +1,132 @@
+"""
+Creates documentation from .py files
+"""
+import os
+import sys
+from os.path import join, abspath, dirname, basename
+import subprocess
+
+FILE_ABS_SOURCE = abspath(dirname(__file__))
+MAIN_PY_SRC = FILE_ABS_SOURCE + "/../source-temp/user/models/src/"
+if os.path.exists(MAIN_PY_SRC):
+    ABSOLUTE_TARGET_MAIN = abspath(join(dirname(__file__), MAIN_PY_SRC))
+    PLUGIN_PY_SRC = abspath(dirname(__file__)) + "/../../../../.sasview/plugin_models/"
+    ABSOLUTE_TARGET_PLUGINS = abspath(join(dirname(__file__), PLUGIN_PY_SRC))
+else:
+    pass
+
+def get_py(directory):
+    for root, dirs, files in os.walk(directory):
+        # Only include python files not starting in '_' (pycache not included)
+        PY_FILES = [join(directory, string) for string in files if not string.startswith("_") and string.endswith(".py")]
+        return PY_FILES
+
+def get_main_docs():
+    """
+Generates string of .py files to be passed into compiling functions
+Future reference: move to main() function?
+    """
+    # The order in which these are added is important. if ABSOLUTE_TARGET_PLUGINS goes first, then we're not compiling the .py file stored in .sasview/plugin_models
+    TARGETS = get_py(ABSOLUTE_TARGET_MAIN) + get_py(ABSOLUTE_TARGET_PLUGINS)
+    print(get_py(ABSOLUTE_TARGET_PLUGINS))
+    base_targets = [basename(string) for string in TARGETS]
+
+    # Removes duplicate instances of the same file copied from plugins folder to source-temp/user/models/src/
+    for file in TARGETS:
+        if base_targets.count(basename(file)) >= 2:
+            TARGETS.remove(file)
+            base_targets.remove(basename(file))
+
+    return TARGETS
+
+def call_regenmodel(filepath, regen_py):
+    """
+    Runs regenmodel.py/regentoc.py (specified in parameter regen_py) with all found PY_FILES
+    """
+    REGENMODEL = abspath(dirname(__file__)) + "/" + regen_py
+    # Initialize command to be executed
+    command = [
+        sys.executable,
+        REGENMODEL,
+    ]
+    # Append each filepath to command individually if passed in many files
+    if type(filepath) == list:
```

Comment:
`isinstance(filepath, list)` is usually preferred over type checking. https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance